[jit] Add dictionary as output of tracer

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#36696 [jit] Add dictionary as output of tracer**
* #36609 [jit] remove Dict iterationOrder and use insertion order

This PR add dictionary as a supported output of tracer under the strict
flag.

Differential Revision: [D21056962](https://our.internmc.facebook.com/intern/diff/D21056962)

wanchaol committed:
[jit] Add dictionary as output of tracer

This PR add dictionary as a supported output of tracer under the strict
flag.

[ghstack-poisoned]

jamesr66a reviewed (approved):
Great, thanks for hanging in there as we resolved the ordering issues. I'm glad we were able to solve a hidden bug in TorchScript as a side effect of this

wanchaol committed:
Update on "[jit] Add dictionary as output of tracer"


This PR add dictionary as a supported output of tracer under the strict
flag.

Differential Revision: [D21056962](https://our.internmc.facebook.com/intern/diff/D21056962)

[ghstack-poisoned]

wanchaol committed:
Update on "[jit] Add dictionary as output of tracer"


This PR add dictionary as a supported output of tracer under the strict
flag.

Differential Revision: [D21056962](https://our.internmc.facebook.com/intern/diff/D21056962)

[ghstack-poisoned]
